File: python/leetcode/list/3289.py
from collections import Counter
from typing import List
def getSneakyNumbers(nums: List[int]) -> List[int]:
  '''
  3289. The Two Sneaky Numbers of Digitville
  
  參數為數值陣列，回傳元素出現次數大於2的元素。
  回傳的陣列每個必須是唯一值
  '''
  # use counter，可計算出現次數
  # c = Counter(nums)
  # ans = []
  # for key, value in c.items():
  #   if value == 2:
  #     ans.append(key)
  # return ans

  # solution 2.更簡潔寫法
  c = Counter(nums)
  return [key for key,value in c.items() if value == 2]

nums = [7,1,5,4,3,4,6,0,9,5,8,2]
# [4,5]
print(getSneakyNumbers(nums))

File: javascript/LeetCode/Array/3289.js
/**
 * 3289. The Two Sneaky Numbers of Digitville
 * 
 * 參數為數值陣列，回傳元素出現次數大於2的元素。回傳的陣列每個必須是唯一值
 * 
 * @param {number[]} nums
 * @return {number[]}
 */
var getSneakyNumbers = function(nums) {
    // element出現次數 >= 2才能被保留
    // 回傳的陣列元素必須是唯一值
    // let map = new Map();
    // let ans = [];
    // for(let i = 0;i < nums.length;++i) {
    //   map.has(nums[i]) ? map.set(nums[i],map.get(nums[i])+ 1) : map.set(nums[i],1);
    // }
    // for(const [key,value] of map) {
    //   if(value >= 2){
    //     ans.push(key);
    //   }
    // }
    // return ans;

    // solution 2.
    let set = new Set();
    let result = [];
    for(let i = 0;i < nums.length;i++) {
      if(set.has(nums[i])){
        result.push(nums[i]);
      }else{
        set.add(nums[i]);
      }
    }
    return result;
};
let nums = [0,1,1,0];
// [0,1]
console.log(getSneakyNumbers(nums));